ERCS/erc-7803.md

```diff
@@ -81,15 +83,27 @@ In the presence of `signingDomains` the account should encode the message to be
 
 `signingDomainSeparators` is the array of hashes of the domains included in `signingDomains`, in the same order, computed according to EIP-712's `hashStruct`.
 
+#### Encoding of signatures for onchain validation
+
+To accommodate existing onchain signature validation functions like [ERC-1271]'s `isValidSignature`, it is possible to use `signature` argument (or equivalent) to ABI encode the signature and its metadata. The following format is RECOMMENDED for completeness:
+
+```solidity
+abi.encodePacked(
+    uint16 signatureLength,              // Length of the actual signature
+    bytes signature,                     // The actual cryptographic signature
+    abi.encode(bytes32[] signingDomainSeparators)  // Signing domain separators (outermost first)
+)
+```
+
 ### `authMethods`
 
 This property is an array of supported signature authentication methods, listed in the order that the verifying domain tries them.
 
 Each member of the array is an object with the following keys:
 
 - `id`: An string that identifies the method. It may be one of:
-    - `ECDSA`: ECDSA signatures by Externally Owned Accounts.
-    - `ERC-{n}`: A standard type of signature specified by an ERC. `n` must not be padded with zeros.
+  - `ECDSA`: ECDSA signatures by Externally Owned Accounts.
+  - `ERC-{n}`: A standard type of signature specified by an ERC. `n` must not be padded with zeros.
```

Comment:
I'm not sure I understand the opposition between ERC-{n} and ECDSA. AFAIK they are two ortogonal things.

- ERC-721 is a way to prepare the hash that is signed, its counterpart is ERC-191
- ECDSA can be used to sign (and verify) signatures, its counterpart would be P256, or ERC-1271.

What does ERC-{n} refers to here, is it 191/721, or is it 1271 ?

Reply:
> its counterpart would be P256, or ERC-1271.

Yes exactly, IIRC this was the reason. It would be one of ERC-1271, ERC-6492, ERC-8010, etc.